opencog/cython/opencog/bindlink.pyx

```diff
@@ -1,12 +1,14 @@
 from opencog.atomspace cimport Atom, AtomSpace, TruthValue
 from opencog.atomspace cimport cHandle, cAtomSpace, cTruthValue
 from opencog.atomspace cimport tv_ptr, strength_t, count_t
+from opencog.atomspace cimport handle_cast
 from cython.operator cimport dereference as deref
 
 def execute_atom(AtomSpace atomspace, Atom atom):
     if atom == None: raise ValueError("execute_atom atom is: None")
-    cdef cHandle c_result = c_execute_atom(atomspace.atomspace,
+    cdef cValuePtr c_value_ptr = c_execute_atom(atomspace.atomspace,
                                            deref(atom.handle))
+    cdef cHandle c_result = handle_cast(c_value_ptr)
     return Atom.createAtom(c_result, atomspace)
```

Comment:
This will work only if the eturned value is actually an atom, otherwise it will crash. If the returned value is a ...value, then the `handle_cast` will return a null, and the `createAtom` will crash. For example the python version of `(cog-execute! (TruthValueOf (Concept "foo")))`   or more generally `(cog-execute! (ValueOf ("Predicate "some-key") (Concept "foo")))` assuming you set the key, previously.

Reply:
`examples/atomspace/stream.scm` for ValueOf

Reply:
`create_value_by_type` can be used instead, see:
https://github.com/opencog/atomspace/blob/5b47b937b7cb1d03785042bf636ae0e2e90a5444/opencog/cython/opencog/atom.pyx#L71-L78